tests/tools/lifecycle.py

```
@@ -94,13 +98,27 @@ async def assert_server_exited_cleanly(server, timeout: float = 10.0) -> None:
+    # Collect stderr AFTER the exit wait: exit-time output (sanitizer
+    # reports, late crash text) must reach the scan below. When a pump is
+    # running it owns the stream — wait for it to see EOF instead of
+    # racing it with a second reader.
     stderr_text = ""
-    if server.stderr:
+    pump = client.stderr_pump if client else None
+    if pump is not None:
+        try:
+            await asyncio.wait_for(asyncio.shield(pump), timeout=2.0)
+        except Exception:
+            pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)